tools/linter/adapters/mypy_linter.py

```diff
@@ -128,8 +132,11 @@ def check_files(
     # file names, see https://github.com/python/mypy/issues/16768
     filenames = [os.path.relpath(f) for f in filenames]
     try:
+        mypy_commands = ["dmypy", "run", "--"]
+        if in_github_actions():
```

Comment:
What if we just use `mypy --fine-grain-cache .`?

Reply:
What effect would that have?  The [docs](https://mypy.readthedocs.io/en/latest/command_line.html#cmdoption-mypy-cache-fine-grained) make it sound like it'll merely add more data to the cache

Reply:
@ZainRizvi It allows mypy to use the same cache as dmypy which should give you most of the same speed up on "warm" mypy caches.